src/rules/no-anonymous-default-export.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @fileoverview Rule to disallow anonymous default exports.
+ * @author Duncan Beevers
+ */
+
+const MAP = {
+  Literal: 'allow-literal',
+  ObjectExpression: 'allow-object-expression',
+  FunctionDeclaration: 'allow-anonymous-function-declaration',
+  ArrowFunctionExpression: 'allow-arrow-function-expression',
+  ClassDeclaration: 'allow-anonymous-class-declaration',
+}
+
+const FLAGS = []
+for (const nodeType in MAP) {
+  FLAGS.push(MAP[nodeType])
+}
+
+module.exports = {
+  meta: {
+    schema: {
+      type: 'array',
+      items: { enum: FLAGS },
+      additionalItems: false,
+      uniqueItems: true,
+    },
+  },
+
+  create: function (context) {
+    const forbid = (option) => context.options.indexOf(option) === -1
+
+    return {
+      'ExportDefaultDeclaration': (node) => {
+        const type = node.declaration.type
+        const noID = !node.declaration.id
+
+        if (type === 'Literal' && forbid(MAP[type])) {
+          context.report({
+            node: node,
+            message: 'Unexpected default export of literal',
+          })
+          return
+        }
+
+        if (type === 'ObjectExpression' && forbid(MAP[type])) {
+          context.report({
+            node: node,
+            message: 'Unexpected default export of object expression',
+          })
+          return
+        }
+
+        if (type === 'FunctionDeclaration' && noID && forbid(MAP[type])) {
+          context.report({
+            node: node,
+            message: 'Unexpected default export of anonymous function',
```

Comment:
I've added articles are all the default export types for consistency.

```
a literal
an array
an object
```
etc...